Rules/UseDeclaredVarsMoreThanAssignments.cs

```diff
@@ -113,7 +114,7 @@ private IEnumerable<DiagnosticRecord> AnalyzeScriptBlockAst(ScriptBlockAst scrip
             IEnumerable<Ast> varAsts = scriptBlockAst.FindAll(testAst => testAst is VariableExpressionAst, true);
             IEnumerable<Ast> varsInAssignment;
 
-            Dictionary<string, AssignmentStatementAst> assignments = new Dictionary<string, AssignmentStatementAst>(StringComparer.OrdinalIgnoreCase);
+            Dictionary<string, AssignmentStatementAst> assignmentsDictionary_OrdinalIgnoreCase = new Dictionary<string, AssignmentStatementAst>(StringComparer.OrdinalIgnoreCase);

```

Comment:
Can you please explain me what the variable name: "assignmentsDictionary_OrdinalIgnoreCase" stands for ?

Reply:
Sure. As per Jim's [comment here](https://github.com/PowerShell/PSScriptAnalyzer/pull/925#discussion_r175177807), I renamed it to make it more obvious that the dictionary is case insensitive (one can see this option only when looking at its constructor). PowerShell is overall case sensitive (except for [this](https://stackoverflow.com/questions/30169887/why-get-itemproperty-is-case-insensitive-but-set-itemproperty-is-case-sensitive) special case), therefore this a standard appoach and `OrdinalIgnoreCase` is used because `InvariantCultureIgnoreCase` is not fully available in netstandard. About the variable itself: the rule first tries to find all variable assignments and adds them to the dictionary, then it looks for all usages of variables and removes the variable from the dictionary if it gets used. The variables that are left must therefore be unused and PSSA returns then the warnings.